crates/data_components/src/object/mod.rs

```diff
@@ -45,7 +45,32 @@ impl ObjectStoreContext {
     ) -> Result<Self, Box<dyn std::error::Error + Send + Sync>> {
         let (prefix, filename_regex_opt) = parse_prefix_and_regex(url, extension)?;
         let filename_regex = filename_regex_opt
-            .map(|regex| Regex::new(&regex).boxed())
+            .map(|regex| {
```

Comment:
Is this for globbing? If so you can use the glob crate, which already has some decent amount of validations for acceptable patterns:

https://docs.rs/glob/latest/glob/struct.Pattern.html

Reply:
We use glob crate for filtering, but this is validation for URLs, so I don't think we can use it.